### ISSUE TITLE
Has stopped working after upgrade 2.0.4 → 2.0.5

### ISSUE THREAD

**Issue by jock-tanner** [aljosa/django-tinymce#132]
Here’s a traceback.

```
/home/tanner/workspace/ROOTWELT/studio/project/env/bin/python3.4 /home/tanner/workspace/ROOTWELT/studio/project/studio/manage.py runserver
Traceback (most recent call last):
  File "/home/tanner/workspace/ROOTWELT/studio/project/studio/manage.py", line 8, in <module>
    execute_from_command_line(sys.argv)
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/site-packages/django/core/management/__init__.py", line 338, in execute_from_command_line
    utility.execute()
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/site-packages/django/core/management/__init__.py", line 312, in execute
    django.setup()
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/site-packages/django/__init__.py", line 18, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/site-packages/django/apps/registry.py", line 85, in populate
    app_config = AppConfig.create(entry)
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/site-packages/django/apps/config.py", line 86, in create
    module = import_module(entry)
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/importlib/__init__.py", line 109, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
  File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
  File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1129, in _exec
  File "<frozen importlib._bootstrap>", line 1471, in exec_module
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "/home/tanner/workspace/ROOTWELT/studio/project/studio/mptt_comments/__init__.py", line 1, in <module>
    from mptt_comments.models import MPTTComment
  File "/home/tanner/workspace/ROOTWELT/studio/project/studio/mptt_comments/models.py", line 8, in <module>
    from tinymce.models import HTMLField
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/site-packages/tinymce/models.py", line 6, in <module>
    from tinymce import widgets as tinymce_widgets
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/site-packages/tinymce/widgets.py", line 10, in <module>
    import tinymce.settings
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/site-packages/tinymce/settings.py", line 17, in <module>
    JS_ROOT = getattr(settings, 'TINYMCE_JS_ROOT', finders.find('tiny_mce', all=False))
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/site-packages/django/contrib/staticfiles/finders.py", line 249, in find
    for finder in get_finders():
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/site-packages/django/contrib/staticfiles/finders.py", line 264, in get_finders
    yield get_finder(finder_path)
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/functools.py", line 448, in wrapper
    result = user_function(*args, **kwds)
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/site-packages/django/contrib/staticfiles/finders.py", line 277, in get_finder
    return Finder()
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/site-packages/django/contrib/staticfiles/finders.py", line 129, in __init__
    app_configs = apps.get_app_configs()
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/site-packages/django/apps/registry.py", line 137, in get_app_configs
    self.check_apps_ready()
  File "/home/tanner/workspace/ROOTWELT/studio/project/env/lib/python3.4/site-packages/django/apps/registry.py", line 124, in check_apps_ready
    raise AppRegistryNotReady("Apps aren't loaded yet.")
django.core.exceptions.AppRegistryNotReady: Apps aren't loaded yet.

Process finished with exit code 1
```

**Comment by Natim** [comment]
Oh can you try to import tinymce after staticfiles?
If you are not using staticfile then we've got to make a fix.
Can you try changing https://github.com/aljosa/django-tinymce/compare/2.0.4...2.0.5#diff-1d5fa3b685a77f9452e28aa136cd8319R3 from line 3 to line 17?

**Comment by jock-tanner** [comment]
Yes I use contrib.staticfiles. Here's what my settings.INSTALLED_APPS looks like:

```
INSTALLED_APPS = [
    'grappelli',
    'modeltranslation',
    'filebrowser',
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django_comments',
    'django.contrib.sites',
    'django.contrib.humanize',
    'rosetta',
    'social.apps.django_app.default',
    'account',
    'django_bleach',
    'tinymce',
    'sorl.thumbnail',
    'mce_filebrowser',
    'sitetree',
    'bootstrapform',
    'pinax_theme_bootstrap',
    'bootstrap_pagination',
    'mathfilters',
    'haystack',
    'mptt',
    'captcha',
    'rest_framework',
    'text_blocks',
    'navigation',
    'blog',
    'bootstrap_calendar',
    'mptt_comments',
]
```

Moving import statement from line 3 to line 17 didn't help (guess it shouldn't in my case).

**Comment by yaniv14** [comment]
I've got the same problem.

**Comment by svfat** [comment]
Same problem

**Comment by vanadium23** [comment]
Is there still an issue with that or are it reproduced on more recent version - 2.2.0.

**Comment by gandalfar** [comment]
I'm trying to use it for a project and I'm seeing this error with 2.2.0. It happens after I do 

`from tinymce.models import HTMLField`
in models.py

Django==1.8.4 and python 3.4

**Comment by Natim** [comment]
This are the changes that occured between 2.0.4 and 2.0.5.

https://github.com/aljosa/django-tinymce/compare/2.0.4...2.0.5

The fact that we use finder at the app loading rathen than latter when the function is used can be the problem here.

**Comment by vanadium23** [comment]
@gandalfar  Can you somehow give a minimal reproducable code example so we can debug it?

**Comment by vanadium23** [comment]
@gandalfar I have provide some [basic app](https://github.com/vanadium23/test-tinymce) for specified version. Can you check it or add what other package do you use?

**Comment by gandalfar** [comment]
Hi, thanks for being so proactive about this issue. I was swamped with organising conference, but I'm now looking through the 100+ packages in requirements.txt to see if I can reproduce this problem.

**Comment by gandalfar** [comment]
I can't reproduce this on a simple app. Will try again later on primary app, to see if I can find cause by removing things instead of adding them.

**Comment by gandalfar** [comment]
Ok, managed to somehow reproduce this. If I comment out 'haystack' from django-haystack==2.4.1 in my project, then it doesn't crash with django-tinymce==2.3.0

I have to do some more work, to get it to crash in your sample app.

**Comment by wyx2000** [comment]
Same problem here,  is there a hotfix for this?

**Comment by Natim** [comment]
@wyx2000 can you update to django-tinymce 2.3.0

**Comment by wyx2000** [comment]
Updated. Still same issue. I have to change the code to remove the line 3 and hardcode the root path


Sent from my Samsung device

-------- Original message --------

**Comment by ayubbeck** [comment]
I had the same problem. Moving my import Model into function fixed the issue!
http://stackoverflow.com/questions/35659282/django-post-migrate-signal-issue

**Comment by adrianmoisey** [comment]
Does anyone have a solution for this issue?

**Comment by adrianmoisey** [comment]
Just tested it, works well 👍

**Comment by Natim** [comment]
Thanks I am going to make a release then.

**Comment by adrianmoisey** [comment]
I do! That would be great if you could do a release.

**Comment by adrianmoisey** [comment]
Thanks! I appreciate it.

**Comment by adrianmoisey** [comment]
Is it possible to get this fix on top of 2.6.0?

**Comment by Natim** [comment]
Is there anything blocking you to use 2.8.0 directly instead?

**Comment by adrianmoisey** [comment]
We're trying to upgrade from Django 1.8.
2.6.0 gives us support for 1.11, and still maintains 1.8 support.

**Comment by adrianmoisey** [comment]
Would you like me to get a 2.6.1 branch going? 
I don't mind doing it.

**Comment by Natim** [comment]
It's fine, I just somehow missed your previous message. Working on it.

**Comment by Natim** [comment]
Testing here: https://travis-ci.org/aljosa/django-tinymce/builds/482310493

**Comment by Natim** [comment]
Released [2.6.1](https://pypi.org/project/django-tinymce/2.6.1/)

**Comment by adrianmoisey** [comment]
Thank you!
